internal/resources/syntheticmonitoring/resource_check.go

```diff
@@ -790,28 +790,26 @@ multiple checks for a single endpoint to check different capabilities.
 		"grafana_synthetic_monitoring_check",
 		resourceCheckID,
 		schema,
-	)
+	).WithLister(listChecks)
 }
 
-// TODO: Fix lister
-// .WithLister(listChecks)
-// func listChecks(ctx context.Context, client *common.Client, data any) ([]string, error) {
-// 	smClient := client.SMAPI
-// 	if smClient == nil {
-// 		return nil, fmt.Errorf("client not configured for SM API")
-// 	}
-
-// 	checkList, err := smClient.ListChecks(ctx)
-// 	if err != nil {
-// 		return nil, err
-// 	}
-
-// 	var ids []string
-// 	for _, check := range checkList {
-// 		ids = append(ids, strconv.FormatInt(check.Id, 10))
-// 	}
-// 	return ids, nil
-// }
+func listChecks(ctx context.Context, client *common.Client, data any) ([]string, error) {
```

Comment:
Out of curiosity, what was it that needed to be fixed - the lister code does not seem to have changed?

Reply:
I don't know. The generator has gone through lots of changes. It works now 🤷